tests/foreman/cli/test_discoveryrule.py

```diff
@@ -505,11 +512,24 @@ def test_positive_update_priority(self):
 
         :CaseImportance: Critical
         """
-        rule = self._make_discoveryrule({u'priority': 100})
-        new_priority = '1'
+        rule_list = DiscoveryRule.list()
+        p_list = []
+        for rule in rule_list:
+            p_list.append(rule['priority'])
+        while True:
+            old_priority = gen_integer(1, 100)
+            if str(old_priority) not in p_list:
+                break
+        rule = self._make_discoveryrule({u'priority': old_priority})
+        self.assertEqual(rule['priority'], str(old_priority))
+        p_list.append(str(old_priority))
+        while True:
+            new_priority = gen_integer(1, 100)
+            if str(new_priority) not in p_list:
+                break
```

Comment:
again, i'd replace all the above added code by this 2-liner:
```python
import random
```

```python
available = set(range(1,1000)) - [r['priority'] for r in DiscoveryRule.list()]
rule_priority = random.sample(available, 1)
```

Reply:
one could use the 1st available value from the set straight away, but i would suggest using the random since otherwise we might hit the conflict with similar tests running in parallel (e.g. 2 tests list the disco rules at the same time, getting the the same p_list would try to create a disco rule with the same priority id again, leading one of them to fail)

Reply:
Thanks @rplevka for your comments. I will do the changes you suggested